Scripts/RemediationScripts/Remediate-ConfigureWAFPolicyForFrontDoor.ps1

```diff
@@ -0,0 +1,600 @@
+                break
+            }
+            else
+            {
+                Write-Host "WAF Policy Mode will be Configured for all Front Door FrontEndpoint(s)" -ForegroundColor $([Constants]::MessageType.Update)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+        else
+        {
+            Write-Host "'Force' flag is provided. WAF Policy will be configured for all Front Door Frontendoint(s) without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host $([Constants]::SingleDashLine)
+        }
+    
+ 
+        Write-Host "[Step 5 of 5] Configuring WAF Policy for Front Door Endpoint(s)"
+        Write-Host $([Constants]::SingleDashLine)
+
+        # To hold results from the remediation.
+        $frontDoorFrontendpointsRemediated = @()
+
+        Write-Host "Configuring WAF Policy on FrontDoor Frontend Endpoint(s)"
+        Write-Host $([Constants]::SingleDashLine)
+
+        # Remidiate Controls by Configuring WAF Policy
+        $frontDoorEndpointsWithWAFPolicyNotConfigured | ForEach-Object {
+            $frontDoorEndPoint = $_
+            $frontdoorName = $_.FrontDoorName;
+            $resourceGroupName = $_.ResourceGroupName; 
+
+            # Holds the list of Front Door Endpoints without WAF Policy Configure
+            $frontendpointsSkipped = @()
+            $frontendpointsSkippedStr = [String]::Empty
+             
+            try
+            {   
+                Do
+                {
+                    $wafPolicyName = Read-Host -Prompt "Enter WAF Policy Name for Endpoint: [$($_.EndPointName)] of Frontdoor [$($frontdoorName)] " 
+                    Write-Host $([Constants]::SingleDashLine)
+                    $policyResourceGroup = Read-Host -Prompt "Enter WAF Policy Resource Group Name for Endpoint: [$($_.EndPointName)] of Frontdoor [$($frontdoorName)] "
+
+                    $wafPolicyName = $wafPolicyName.Trim()
+                    $policyResourceGroup = $policyResourceGroup.Trim()
+
+                    Write-Host $([Constants]::SingleDashLine)
+                    $policy = Get-AzFrontDoorWafPolicy -Name  $wafPolicyName  -ResourceGroupName $policyResourceGroup -ErrorAction SilentlyContinue
+
+                    if($policy -eq $null)
+                    {
+                        Write-Host "WAF Policy name or WAF Policy Resource Group Name is not correct. Please enter correct details."
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+
+                    if($policy -ne $null -and $policy.Sku -ne 'Classic_AzureFrontDoor')
+                    {
+                        Write-Host "WAF Policy is not of type Front door tier Classic . Please enter correct details."
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+                }
+                while($policy.Sku -ne 'Classic_AzureFrontDoor' -or $policy -eq $null)
```

Comment:
Suppose if user don't have WAF policy resource for current front door then user will be stuck. We should give some force exit option to user.